src/vs/workbench/contrib/terminal/browser/media/shellIntegration-login.zsh

```diff
@@ -0,0 +1,9 @@
+# ---------------------------------------------------------------------------------------------
+#   Copyright (c) Microsoft Corporation. All rights reserved.
+#   Licensed under the MIT License. See License.txt in the project root for license information.
+# ---------------------------------------------------------------------------------------------
+if [[ $options[norcs] = off && -o "login" &&  -f $USER_ZDOTDIR/.zlogin ]]; then
+	. $USER_ZDOTDIR/.zlogin
```

Comment:
in case this file references ZDOTDIR, set it to the user's ZDOTDIR and then back. do this for the other scripts too and then can remove line 9